mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/impl/ClientState.scala

```diff
@@ -151,32 +154,58 @@ import scala.util.{Failure, Success}
+                                          connect: Connect,
                                           connectData: ConnectData,
                                           remote: Promise[Source[ForwardConnectCommand, NotUsed]])
-      extends Event
-  final case class ConnAckReceivedFromRemote(connAck: ConnAck, local: Promise[ForwardConnAck]) extends Event
-  case object ReceiveConnAckTimeout extends Event
-  case object ConnectionLost extends Event
-  final case class DisconnectReceivedLocally(remote: Promise[ForwardDisconnect.type]) extends Event
-  final case class SubscribeReceivedLocally(subscribe: Subscribe,
+      extends Event(connectionId)
+  final case class ConnAckReceivedFromRemote(override val connectionId: ByteString,
+                                             connAck: ConnAck,
+                                             local: Promise[ForwardConnAck])
+      extends Event(connectionId)
+
+  case class ReceiveConnAckTimeout(override val connectionId: ByteString) extends Event(connectionId)
+
+  case class ConnectionLost(override val connectionId: ByteString) extends Event(connectionId)
+
+  final case class DisconnectReceivedLocally(override val connectionId: ByteString,
+                                             remote: Promise[ForwardDisconnect.type])
+      extends Event(connectionId)
+
+  final case class SubscribeReceivedLocally(override val connectionId: ByteString,
+                                            subscribe: Subscribe,
                                             subscribeData: Subscriber.SubscribeData,
                                             remote: Promise[Subscriber.ForwardSubscribe])
-      extends Event
-  final case class PublishReceivedFromRemote(publish: Publish, local: Promise[Consumer.ForwardPublish.type])
-      extends Event
-  final case class ConsumerFree(topicName: String) extends Event
-  final case class PublishReceivedLocally(publish: Publish, publishData: Producer.PublishData) extends Event
-  final case class ProducerFree(topicName: String) extends Event
-  case object SendPingReqTimeout extends Event
-  final case class PingRespReceivedFromRemote(local: Promise[ForwardPingResp.type]) extends Event
-  final case class ReceivedProducerPublishingCommand(command: Source[Producer.ForwardPublishingCommand, NotUsed])
-      extends Event
-  final case class UnsubscribeReceivedLocally(unsubscribe: Unsubscribe,
+      extends Event(connectionId)
+
+  final case class PublishReceivedFromRemote(override val connectionId: ByteString,
+                                             publish: Publish,
+                                             local: Promise[Consumer.ForwardPublish.type])
+      extends Event(connectionId)
+
+  final case class ConsumerFree(topicName: String) extends Event(ByteString.empty)
+
+  final case class PublishReceivedLocally(publish: Publish, publishData: Producer.PublishData)
+      extends Event(ByteString.empty)
```

Comment:
These and related aren't associated with a connection id, so their connection id is an empty `ByteString`. This is similar to how the server does things.